src/main/scala/org/broadinstitute/dsde/firecloud/core/AgoraPermissionHandler.scala

```diff
@@ -0,0 +1,307 @@
+package org.broadinstitute.dsde.firecloud.core
+
+import akka.actor.{Actor, Props}
+import akka.event.Logging
+import org.broadinstitute.dsde.firecloud.model.MethodRepository.{AgoraPermission, FireCloudPermission}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.service.FireCloudRequestBuilding
+import org.broadinstitute.dsde.firecloud.service.PerRequest.RequestComplete
+import org.broadinstitute.dsde.firecloud.HttpClient
+import spray.client.pipelining._
+import spray.http.StatusCodes._
+import spray.http.{Uri, HttpResponse, StatusCodes}
+import spray.httpx.SprayJsonSupport._
+import spray.json.DefaultJsonProtocol._
+import spray.routing.RequestContext
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+object AgoraPermissionHandler {
+  case class Get(url: String)
+  case class Post(url: String, agoraPermissions: List[AgoraPermission])
+  def props(requestContext: RequestContext): Props = Props(new GetEntitiesWithTypeActor(requestContext))
+
+  def testFromFCToAgora():Boolean = {
```

Comment:
test code should live only within the test package; there should be no test-only code in runtime classes.